Objects/obmalloc.c

```diff
@@ -979,10 +986,8 @@ static inline OMState *
 get_state(void)
 {
     PyInterpreterState *interp = _PyInterpreterState_GET();
-    if (!has_own_state(interp)) {
-        interp = _PyInterpreterState_Main();
-    }
-    return &interp->obmalloc;
+    assert(interp->obmalloc != NULL); // otherwise not initialized or freed
+    return interp->obmalloc;
```

Comment:
The existing code is still correct, isn't it?

```suggestion
    if (!has_own_state(interp)) {
        assert(interp->obmalloc == NULL);
        interp = _PyInterpreterState_Main();
    }
    assert(interp->obmalloc != NULL); // otherwise not initialized or freed
    return interp->obmalloc;
```

Alternately, we don't need to bother setting `interp->obmalloc` for the main interpreter:

```suggestion
    if (_Py_IsMainInterpreter(interp)) {
        assert(interp->obmalloc == NULL);
        return &obmalloc_state_main;
    }
    else if (_PyInterpreterState_HasFeature(interp, Py_RTFLAGS_USE_MAIN_OBMALLOC)) {
        assert(interp->obmalloc == NULL);
        return &obmalloc_state_main;
    }
    assert(interp->obmalloc != NULL); // otherwise not initialized or freed
    return interp->obmalloc;
```

Is the intention to avoid the branch for a little performance?  If so, how much does it win us?  I'd expect the branch predictor to help out somewhat.

Reply:
Yes, the idea is to avoid branches.  The difference is probably tiny but this is a quite a hot path.